### DISCUSSION TITLE
Welcome to ProtocolWorkingGroup Discussions!

### DISCUSSION THREAD

**Discussion by mcsadmin** [CredComSoc/ProtocolWorkingGroup#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to discuss and decide how best to guide the development of the Credit Commons Protocol. 

A good place to start might be the [Where is the protocol at and where do we need it to be?](https://github.com/CredComSoc/ProtocolWorkingGroup/discussions/3) discussion.

To understand why we're doing this, and how we are approaching it, the [Direction, Vision and Ethics](https://github.com/CredComSoc/ProtocolWorkingGroup/discussions/2) discussion should be useful.

We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JohnG-cc** [comment]
Hi, I'm John. I'm a software developer with an interest in alternative currency, pro-competitive business structures, and industrial democracy. I have been talking with Dil, Matthew, and Tom W. about helping with software development related to the Credit Commons Protocol. 

I wasn't able to make it to Austria for the Collaborative Finance conference, but I very much enjoyed watching all the live streams on Youtube. I have also been reading through and playing around with the protocol reference implementation code on GitLab to get an idea of how it works under the hood.

**Comment by dilgreen** [comment]
Hi, I'm Dil, former architect and builder, turned .. what? 

Well, we're homing in on something around 'meso-economic architect' 'meso-economy designer'. Who knows? We recently came across some design theorists talking about 'Transition design', and we like this too - as it brings in the requirement for systems-change understanding. 'Radical Generalists' was another term they introduced.

The thing being, to be able to design, and help people build social scale economic contexts - markets of one kind or another, which have their own 'inside' money, which is connected to other moneys, of wider and narrower scope, in such a way that it retains its integrity, its sense of self, without losing its connection to other markets.

The purpose of this work being to support the development of a 'life-like economy' - one which supports and encourages full engagement from its participants - including their feelings and their ethics, in the conviction that it is only through such integration that the abstract number world of money healthily interact with the biosphere. That there is no sane direct interaction - only interactions mediated by humans in sociality.

**Comment by utunga** [comment]
Yo ! I'm Miles aka @utunga

I'm really loving these ideas of 'life like' economies and full engagement with ethics and biosphere in the way we design monetary systems.

And also @OtherJohnGray-CreditCommons man I feel totally feel you about that conference. That looked totally *amazing*. I literally was just writing to a friend..  
> These are the videos from the Collaborative Finance hui that they had in the Austria recently, that i mentioned.
> https://www.youtube.com/@cryptocommonsassociation2609/streams

Huge props to @matslats for bringing together that crew of people. (And the concept/format of a single track, 5 day conference with stacks of time for chats in between as way to address this kind of material - *chefs kiss*) Total #FOMO. Anyway.. But yeah it is cool that we had the live streams for us remote people.

Actually i swear I was thinking before I got here today that a lovely foundational concept for the mahi (work) that I hope we're undertaking here .. is.. this bit from tha conference where @matslats is talking.  (OK Hang on this is gonna take a minute to find.. OMG grassroots! I have so much stuff to catch up on! anyway)

https://www.youtube.com/watch?v=kXLisL03nvk&t=4176s 

I took the liberty of transcribing it here:

> What tends to happen is someone like me comes up with an absolutely brilliant idea and I stayed in my bedroom for several years and I built this software and then tried tget everybody to use it, but it's still only one software, it's one silo. It doesn't fit.
>
> It's me trying to fix all of the problems of the world with one tool and it's just the wrong way to think.
>
> Sure! Build your systems. Experiment! It's fantastic learning. The software led me into the monetary theory and made me the person I am today.
>
> But I long ago stopped trying to be the one building the silos, and we shouldn't even be thinking in terms of one system because it's not reslient
>
> And this is what happens in silicon vallye. The model is that you get a monopoly and what the venture capitalists do is, they fund you on the hope and the probability that you will get a monopoly and capture a market.
>
> But that's not going to save us either.
>
> I mean, even if you could get them on, even if you couldn't convince them that your system is going to go viral and capture all the users and *suck the money out of them*. Even if you do that, it's not going to fix it in a way that I would recognise. So I've had very little engagement with that kind of venture capital scene and consequently remained poor through throughout my 40s.

I feel like it sets the scene really nicely for me. 

--

Actually a ton of what Mats said - in general scene setting for that conference - is stuff that I feel really nicely sets the scene for what we are/were already hoping to do here. 

For instance this earlier bit where @matslats was sort of[ introducing the four diferent models for "structures of credit"](https://youtu.be/kXLisL03nvk?t=4062). Is also a great scene setter.

It feels like a part of the *protocol design* work we are hoping to do here might involve creating way for meshing together or melding those different models together. (I am especially keen on merging mutual credit and multilateral offset clearing into one conception - also perhaps equity and mutual staking etc could be involved).

_I also really wanna explore how tokenized and relationship / credit forms of 'money' - for want of better word - can work together in one system. Which is in some ways related to thinking about how mutual credit is a form of liquidity injection into mtcs.. but anyway_

Also just to put it out there I think an important part of the work in protocol designs might be *simulations*. That s I absolutely think we should do them right from the start and we should do lots of them. Well that's part of what I'm going to try and do anyway. It seems to me that even just _understanding_ and _discussing_ how such systems *might* work is a tricky matter sometimes. They kind of work as cybernetic systems with potentially quite complex  state spaces. So yeah absolutely do think its worth the effort to be clear enough and to create visualizations as a form of communication of how such systems might work. 

[React-Flow](https://reactflow.dev/) seems like a cool way  to do such work and I could imagine embedding it in html documents like notebooks. Or we could also use vis.js but of the two I'd def choose React Flow at this point.

And other ideas also... but for now just glad to be here. Look to my @utunga bio to get a sense of who I am (if you havent done that already) and looking forward to talking more.

**Comment by JohnG-cc** [comment]
P.S. I just shortened my username - OtherJohnGray-CreditCommons was ridiculously long for 'at' usage!

**Comment by bu6n** [comment]
Hi! I'm Billy from Switzerland. After having met almost all of you in different places and chats (the Austria conference, Telegram, etc.), I thought it would be a good idea to introduce myself officially here.

I've been involved in a local group for almost 2 years here, searching for good alternative currencies design and ways to implement them. This led us to mutual credit, then to the Austria conference and now we think we have found what we want to do. At the moment we are trying to set up something similar to local loop and in a second step we want to introduce mutual credit. That's what led me to this discussion group. My bachelor thesis, which is due by March 2024, will research the security of the current reference implementation and one of the goals is to also make code contributions. I'm also interested in documentation, since I will have to explain the protocol succinctly in the paper. 

I can provide help to setup automatic documentation from a repo in GitLab or GitHub if needed. The best would be to have Markdown files. I think that's what most people are used to read and modify.

@utunga I feel you on the simulation side. The Holochain project has a very nice playground: https://holochain-playground.github.io/
